Add GetUnityVersion support for Unity 5.4 and lower

spacehamster opened the pull request:
`?GameEngineVersion@PlatformWrapper@UnityEngine@@SAPEBDXZ` is not available on Unity 5.3 and lower.

`Application_Get_Custom_PropUnityVersion` might be a good candidate for the new regex symbol resolver, I wrote it the way I did to list out all the specific versions.

Unity 3.4 to 4.7
```
?Application_Get_Custom_PropUnityVersion@@YAPAUMonoString@@XZ
struct MonoString * __cdecl Application_Get_Custom_PropUnityVersion(void)
```
5.0 to 2019.4
```
Application_Get_Custom_PropUnityVersion@@YAPEAUMonoString@@XZ
struct MonoString * __cdecl Application_Get_Custom_PropUnityVersion(void)
```
Unity 2020.1 onwards
```
?Application_Get_Custom_PropUnityVersion@@YAPEAVScriptingBackendNativeStringPtrOpaque@@XZ
class ScriptingBackendNativeStringPtrOpaque * __cdecl Application_Get_Custom_PropUnityVersion(void)
```

Should `?GameEngineVersion@PlatformWrapper@UnityEngine@@SAPEBDXZ` continue to be used for versions 5.5 or later or completely replaced?

Should there be a separate class for resolving mono functions? I'm not sure if it will come up much.

Sidenote: I'm not a fan of manually escaping regex like in the example
```
resolver.ResolveFirstFunctionMatching<CStrDelegate>(new Regex(@"\?c_str@\?\$basic_string@*"));
``` 
I think this would be better
```
resolver.ResolveFirstFunctionMatching<CStrDelegate>(new Regex(Regex.Escape("?c_str@?$basic_string@") + "*")));
```

DaZombieKiller commented:
> Should `?GameEngineVersion@PlatformWrapper@UnityEngine@@SAPEBDXZ` continue to be used for versions 5.5 or later or completely replaced?

I think we should continue to use that whenever it's available.

> Should there be a separate class for resolving mono functions? I'm not sure if it will come up much.

Probably not, but if it does end up becoming more frequent later we can add one.

> Sidenote: I'm not a fan of manually escaping regex like in the example

Yeah, I just completely forgot about `Regex.Escape` when I wrote that.

spacehamster commented:
I've added `?GameEngineVersion@PlatformWrapper@UnityEngine@@SAPEBDXZ` back in. It turns out `??0UnityVersion@@QEAA@PEBD@Z` is only available starting from Unity 5.4 and later, so i added string parsing to UnityVersion for 5.3 and earlier.
